### PR TITLE
Bug 1311980 - Gracefully handle invalid job_id in BugJobMapViewSet.list

### PR DESCRIPTION
**Pull request by alisha17**
The value error is now handled gracefully with HTTP 400.
Bugzilla link: https://bugzilla.mozilla.org/show_bug.cgi?id=1311980

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/2142)

### CODE REVIEW

File: treeherder/webapp/api/bug.py [hunk 1]
```diff
@@ -55,14 +55,19 @@ def retrieve(self, request, project, pk=None):
...
-        serializer = BugJobMapSerializer(bug_job_maps, many=True)
+            jobs = Job.objects.filter(repository__name=project, id__in=job_ids)
+            bug_job_maps = BugJobMap.objects.filter(job__in=jobs).select_related(
+                'user')
+            serializer = BugJobMapSerializer(bug_job_maps, many=True)
 
-        return Response(serializer.data)
+            return Response(serializer.data)
+
+        except ValueError:
```
Comment (edmorley) [inline treeherder/webapp/api/bug.py]: Could you narrow the scope of the try-except, such that it just wraps the `map()` line? This should improve readability of the code (and also prevent catching cases that are unrelated).

Could you also add a test to:
https://github.com/mozilla/treeherder/blob/master/tests/webapp/api/test_bug_job_map_api.py

Many thanks!

File: tests/webapp/api/test_bug_job_map_api.py [hunk 2]
```diff
@@ -148,3 +162,17 @@ def test_bug_job_map_delete(webapp, eleven_jobs_stored, test_repository,
...
+
+
+def test_bug_job_bad_job_id(webapp, test_repository):
+    """
+    test we have graceful error when we pass an invalid job_id
+    """
+    bad_job_id = "aaaa"
+
+    resp = webapp.get(
+            reverse("bug-job-map-list", kwargs={"project": test_repository.name}),
```
Comment (edmorley) [inline tests/webapp/api/test_bug_job_map_api.py]: Thank you for adding a test to cover this other case too - however I think it should be using `bug-job-map-detail` not `bug-job-map-list` (see example in `test_bug_job_map_delete`). With that changed, the test will (correctly) start failing, since `BugJobMapViewSet.retrieve()` doesn't yet handle the `ValueError` gracefully (could you add the check there too?).

Comment (alisha17) [inline tests/webapp/api/test_bug_job_map_api.py]: I suppose the changes which you have suggested are not related to this bug. But I would fix this also in a seperate bug/PR.

Comment (edmorley) [inline tests/webapp/api/test_bug_job_map_api.py]: Yeah agree a new PR would be preferable. I only mentioned it since this newly added test was testing behaviour unrelated to the rest of the PR (and with a typo).

Comment (edmorley) [inline tests/webapp/api/test_bug_job_map_api.py]: The PR currently fixes one issue (in the `list` view for the bug-job-map API), but adds two tests, when I would have expected just one. 

The extra test (`test_bug_job_map_list_bad_job_id`) has a name that implies it's trying to test the `detail` view for the bug-job-map API, but if that's the case:
(a) the contents of the test aren't correct, since it's actually duplicating `test_bug_job_bad_job_id`, when it would need to be using `reverse("bug-job-map-detail", ..."` instead.
(b) then the PR would also need to be updated, since the detail view is actually broken currently (ie needs the try-except too).

So therefore the options are to either:
(a) remove the duplicate test and open a new bug/PR
(b) fix the test and the detail view in the same PR

I initially proposed (b), but after you mentioned wanting to do this in another PR was saying it's fine to do (a).

Comment (alisha17) [inline tests/webapp/api/test_bug_job_map_api.py]: Ahh.. Got it know! Will fix this and create a new PR for detail view bug. Thanks!

File: tests/webapp/api/test_bug_job_map_api.py [hunk 3]
```diff
@@ -148,3 +148,17 @@ def test_bug_job_map_delete(webapp, eleven_jobs_stored, test_repository,
         content = json.loads(resp.content)
         assert content == {"message": "Bug job map deleted"}
         assert BugJobMap.objects.count() == 0
+
+
+def test_bug_job_bad_job_id(webapp, test_repository):
```
Comment (edmorley) [inline tests/webapp/api/test_bug_job_map_api.py]: Perhaps name the test `test_bug_job_map_bad_job_id`?

Other than that looks great - I'll merge once the test name is updated :-)
